### PR TITLE
Add linting to build

### PR DESCRIPTION
**Pull request by KamuelaFranco**
* Fixed issues related to linting
* Added linting to build pipeline
* Added dependencies for react-testing-library in preparation for testing

### CODE REVIEW

File: src/components/PreviousSettlementWindowInfo.js [hunk 1]
```diff
@@ -104,7 +101,13 @@ function PreviousSettlementWindowInfo(props) {
 }
 
 PreviousSettlementWindowInfo.propTypes = {
-  previousSettlementWindow: PropTypes.object.isRequired,
+  classes: PropTypes.objectOf({ root: PropTypes.string }).isRequired,
+  previousSettlementWindow: PropTypes.objectOf({
+    payments: PropTypes.array,
+    receipts: PropTypes.array,
+    limits: PropTypes.array,
```
Comment (msk-) [inline src/components/PreviousSettlementWindowInfo.js]: I don't think these are arrays. `Object.keys` is being called on them.

File: src/api.js [hunk 2]
```diff
@@ -24,7 +21,7 @@ async function getCurrentWindow(participantId, requestOpts) {
 }
 
 async function getPreviousWindow(participantId, requestOpts) {
-  return get(`previous-window/${participantId}`);
+  return get(`previous-window/${participantId}`, requestOpts);
```
Comment (Kalaston) [inline src/api.js]: is this a necessary change? what would be the `requestOpts` in this scenario?

File: src/components/SettlementsList.js [hunk 3]
```diff
@@ -59,7 +58,7 @@ function SettlementsList(props) {
     getSettlements(fsp, dates, { ftc })
       .then(setSettlements)
       .catch(ftc.ignoreAbort())
-      .catch(err => window.alert('Failed to get settlement list')); // TODO: better error message, let user retry
+      .catch(() => window.alert('Failed to get settlement list')); // TODO: better error message, let user retry
```
Comment (msk-) [inline src/components/SettlementsList.js]: As we're here, could we console log this error instead?
```suggestion
      .catch((err) => {
        window.alert('Failed to get settlement list'); // TODO: better error message, let user retry
        console.log('Failed to get settlement list', err);
      });
```

File: src/components/SettlementsList.js [hunk 4]
```diff
@@ -72,8 +71,17 @@ function SettlementsList(props) {
   );
 }
 
+SettlementsListList.propTypes = {
+  fspNamesById: PropTypes.objectOf({ id: PropTypes.string }).isRequired,
```
Comment (msk-) [inline src/components/SettlementsList.js]: Is `fspNamesById` used?

Comment (msk-) [reply src/components/SettlementsList.js]: (And if not, I'm surprised linting didn't catch that)

File: src/components/TransactionAverage.js [hunk 5]
```diff
@@ -96,9 +92,9 @@ function TransactionAverage(props) {
   useEffect(() => {
     const ftc = fetchTimeoutController();
     getHistoricalData(fsp.name, dates, { ftc })
-      .then(data => setAverages(data.average))
+      .then((data) => setAverages(data.average))
       .catch(ftc.ignoreAbort())
-      .catch((err) => {
+      .catch(() => {
```
Comment (msk-) [inline src/components/TransactionAverage.js]:
```suggestion
      .catch((err) => {
        console.log('Failed to get averages', err);
```

File: src/components/SettlementsList.js (hunk 4 above)
Comment (KamuelaFranco) [reply src/components/SettlementsList.js]: Yes, it's used.

**Pull request merged**